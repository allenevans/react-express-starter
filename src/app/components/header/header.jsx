/* eslint-disable no-unused-vars,class-methods-use-this */
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import './header.scss';

export class Header extends Component {
  render() {
    return (
      <header>
        <nav>
          <a href="/" className="home-button header-button header-button-small">
            Header
          </a>
        </nav>
      </header>
    );
  }
}

const mapStateToProps = state => ({
  actions: null,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ }, dispatch),
});

Header.propTypes = {
  actions: PropTypes.object.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
