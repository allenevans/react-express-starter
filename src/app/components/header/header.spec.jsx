import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { Header } from './header';

const renderer = new ReactShallowRenderer();

describe('header component', () => {
  it('should render the component', () => {
    const actions = {};

    const tree = renderer.render(<Header actions={actions}/>);

    expect(tree).toMatchSnapshot();
  });
});
