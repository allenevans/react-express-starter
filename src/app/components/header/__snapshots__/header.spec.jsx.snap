// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component should render the component 1`] = `
<header>
  <nav>
    <a
      className="home-button header-button header-button-small"
      href="/"
    >
      Header
    </a>
  </nav>
</header>
`;
